src/bf_dpu_update.py

```diff
@@ -662,17 +692,46 @@ def _wait_task(self, task_handle, max_second=15*60, check_step=10, err_handler=N
             raise Err_Exception(Err_Num.TASK_FAILED, task_state['message'])
         return True
 
+    def validate_args(self, items):
+        if 'UserName' in items:
+            if self.username is None:
+                raise Err_Exception(Err_Num.USERNAME_NOT_GIVEN)
+        if 'Password' in items:
+            if self.password is None:
+                raise Err_Exception(Err_Num.PASSWORD_NOT_GIVEN)
+        if 'BmcIP' in items:
+            if self.bmc_ip is None:
+                raise Err_Exception(Err_Num.BMC_IP_NOT_GIVEN)
+        if 'Module' in items:
+            if self.module is None:
+                raise Err_Exception(Err_Num.MODULE_NOT_GIVEN)
+        if 'FwFile' in items:
+            if self.fw_file_path is None:
+                raise Err_Exception(Err_Num.FW_FILE_NOT_GIVEN)
+            if not os.access(self.fw_file_path, os.R_OK):
+                raise Err_Exception(Err_Num.FILE_NOT_ACCESSIBLE, 'Firmware file: {}'.format(self.fw_file_path))
+        if 'FRU' in items:
+            if not self.oem_fru:
+                raise Err_Exception(Err_Num.FRU_NOT_GIVEN)
+
+        if self.log_file is not None and not os.access(self.log_file, os.W_OK) and not os.access(os.path.abspath(os.path.dirname(self.log_file)), os.W_OK):
+            raise Err_Exception(Err_Num.FILE_NOT_ACCESSIBLE, 'Log file: {}'.format(self.log_file))
+
 
     def validate_arg_for_update(self):
-        if any(v is None for v in (self.username, self.password, self.fw_file_path, self.module, self.bmc_ip)):
-            raise Err_Exception(Err_Num.ARG_FOR_UPDATE_NOT_GIVEN)
+        self.validate_args(['UserName', 'Password', 'BmcIP', 'Module', 'FwFile'])
 
-        if not os.access(self.fw_file_path, os.R_OK):
-            raise Err_Exception(Err_Num.FILE_NOT_ACCESSIBLE, 'Firmware file: {}'.format(self.fw_file_path))
 
-        if self.log_file is not None and not os.access(self.log_file, os.W_OK) and not os.access(os.path.abspath(os.path.dirname(self.log_file)), os.W_OK):
-            raise Err_Exception(Err_Num.FILE_NOT_ACCESSIBLE, 'Log file: {}'.format(self.log_file))
-        return True
+    def validate_arg_for_fru(self):
+        self.validate_args(['UserName', 'Password', 'BmcIP', 'Module', 'FRU'])
+
+
```

Comment:
empty line

Reply:
It's code convention in this repo.

Two lines between functions.  Then we can add one line in functinos.